### ISSUE TITLE
`modernize-use-emplace` not reported with alias

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#55856]
```cpp
#include <list>
#include <string>

typedef std::list<std::string> L1;
using L2 = std::list<std::string>;

void f1()
{
    L1 l;
    l.push_back(""); // no warning
}

void f2()
{
    L2 l;
    l.push_back(""); // no warning
}

void f3()
{
    std::list<std::string> l;
    l.push_back(""); // warning
}
```

https://godbolt.org/z/K5r83c65o

**Comment by JonasToth** [comment]
https://github.com/llvm/llvm-project/blob/8da5d5dbdc0872d3cdbeb5ff42c94775a98c71bc/clang-tools-extra/clang-tidy/modernize/UseEmplaceCheck.cpp#L132

this line needs to match on typedefs as well. i guess `hasCanonicalType` would do that.

**Comment by firewave** [comment]
This also prevents the `unnecessary temporary object` version of this check:

```cpp
#include <list>
#include <string>

typedef std::list<std::string> L1;
using L2 = std::list<std::string>;

void f1()
{
    L1 l;
    l.emplace_back(std::string("")); // no warning
}

void f2()
{
    L2 l;
    l.emplace_back(std::string("")); // no warning
}

void f3()
{
    std::list<std::string> l;
    l.emplace_back(std::string("")); // warning
}
```
https://godbolt.org/z/GEnGoWrjr

**Comment by corona10** [comment]
Still reproduced, I am working on this issue.

**Comment by corona10** [comment]
I submitted the patch: https://reviews.llvm.org/D132640

**Comment by corona10** [comment]
https://github.com/llvm/llvm-project/commit/495d984e14bd8367017ffdea8183840c8267cbbf
Now we can close this issue.
